### PR TITLE
Custom login

### PR DESCRIPTION
**Pull request by ozzycodes2**
Added Providers to Login through Firebase.

Added Crashlytics and Analytics to code.  Validated to display in Firebase console for iOS and Android.

### CODE REVIEW

**Comment by croxx219** [comment]
Don't merge until Adrian's PR goes through.

File: lib/main.dart [hunk 1]
```diff
@@ -1,111 +1,54 @@
...
+import 'package:logger/logger.dart';
+import 'package:provider/provider.dart';
 
-void main() => runApp(MyApp());
+import 'home_page.dart';
+import 'login_page.dart';
+
+void main() =>
+    runApp(ChangeNotifierProvider<AuthService>(
+        child: MyApp(),
```
Comment (AdrianB1995) [inline lib/main.dart]: Introduce a router with initial route set to the login_page.

**Comment by AdrianB1995** [comment]
Pull from master to resolve conflicts.

File: lib/main.dart (hunk 1 above)
Comment (ozzycodes2) [reply lib/main.dart]: Done. Using a different approach but should work all the same

**Review by AdrianB1995 — APPROVED** [review]
Looks good!

**Pull request merged**